src/VirtoCommerce.MarketingModule.Data/Repositories/MarketingRepository.cs

```diff
@@ -177,6 +178,25 @@ public Task RemoveMarketingUsagesAsync(string[] ids)
         }
         #endregion
 
+        public async Task<string[]> CheckCouponsForUniquenessAsync(Coupon[] coupons)
+        {
+            var result = new List<string>();
+            var uniqueListCodeAndPromotionId = coupons.Select(x => x.Code + x.PromotionId).Distinct().ToArray();
+
+            for (var skip = 0; skip < uniqueListCodeAndPromotionId.Length; skip += DefaultPageSize)
+            {
+                var part = uniqueListCodeAndPromotionId.Skip(skip).Take(DefaultPageSize).ToArray();
+                var errors = await Coupons.Where(x => part.Contains(x.Code + x.PromotionId))
+                    .Include(x => x.Promotion)
+                    .Select(x => $"Coupon with Name: '{x.Code}' for Promotion: '{x.Promotion.Name}' already exists;")
```

Comment:
I think semicolon at the end of the message is redundant: "exists**;**"

Reply:
Without any ending in an error message would be difficult to separate one message from another one. Might we use dot instead of semicolon? 
![2020-04-07-10-52_ыуьшсщдщт](https://user-images.githubusercontent.com/11813599/78644332-62819400-78be-11ea-95ce-23ab7be7af79.png)

Reply:
Changed to dot.